lib/Service/StackService.php

```diff
@@ -362,4 +379,106 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @param $userId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId, $userId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+		if (is_numeric($boardId) === false) {
+			throw new BadRequestException('board id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_READ);
+		if ($this->boardService->isArchived(null, $boardId)) {
+			throw new StatusException('Operation not allowed. This board is archived.');
+		}
+
```

Comment:
:warning: We also need to verify that the stack is actually part of the board, otherwise this would be a security issue where an attacker could clone a stack of another user to their own board.

Reply:
Please do not resolve this comment, I'll do that once verified that this case is covered. :wink:

Reply:
Could you also address that?